pkg/api/v20230701preview/openshiftcluster_validatestatic.go

```diff
@@ -228,6 +231,44 @@ func (sv openShiftClusterStaticValidator) validateNetworkProfile(path string, np
 			return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, path+".outboundType", "The provided outboundType '%s' is invalid: cannot use UserDefinedRouting if either API Server Visibility or Ingress Visibility is public.", np.OutboundType)
 		}
 	}
+
+	if np.OutboundType == OutboundTypeUserDefinedRouting && np.LoadbalancerProfile != nil {
+		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, path+".loadBalancerProfile", "The provided loadBalancerProfile is invalid: cannot use a loadBalancerProfile if outboundType is UserDefinedRouting.")
+	}
+
+	return nil
+}
+
+func (sv openShiftClusterStaticValidator) validateLoadBalancerProfile(path string, lbp *LoadbalancerProfile, isCreate bool) error {
+	if lbp != nil {
+		var isManagedOutboundIPCount = lbp.ManagedOutboundIPs != nil
+		var isOutboundIPs = lbp.OutboundIPs != nil
+		var isOutboundIPPrefixes = lbp.OutboundIPPrefixes != nil
+
+		if isManagedOutboundIPCount && !isOutboundIPPrefixes && !isOutboundIPs {
+			if !(lbp.ManagedOutboundIPs.Count > 0 && lbp.ManagedOutboundIPs.Count <= 20) {
```

Comment:
nit: combine this condition with the line above to avoid single path condition nesting.